addon/mixins/admin-adapter-mixin.js

```diff
@@ -0,0 +1,72 @@
+import Ember from 'ember'
+import HasManyQuery from 'ember-data-has-many-query'
+
+const {
+  Mixin
+} = Ember
+const {
+  RESTAdapterMixin
+} = HasManyQuery
+
+export default Mixin.create(RESTAdapterMixin, {
+  removeRelated(model, relationshipName, relatedModel, kind) {
+    let verb
+    let data
+    if (kind === 'belongsTo') {
+      verb = 'PATCH'
+      data = null
+    } else {
+      verb = 'DELETE'
+      data = [{
+        type: this._payloadKeyFromModel(relatedModel),
+        id: relatedModel.get('id')
+      }]
+    }
+
+    return this.ajax(
+      this.urlForRemoveFromHasMany(model, relationshipName),
+      verb,
+      { data: { data }}
```

Comment:
Just to clarify, is the first `data` key specifying stuff that goes into the body of the request, and the second data key the json that actually gets sent to the server?

I know this will expand to:

```
{ data: { data: null }}
```

or

```
{ data: { data: [{ ... }] }}
```

I'm not entirely sure of the `this.ajax` API, but just wanted to be sure this wouldn't send the body as `{ data: { data: <stuff> }}`

Reply:
The request only has a single `data` key. I was surprised I had to nest the key twice as well.

```js
{ data: { type: 'resources', ... }}
```